sopel/modules/admin.py

```diff
@@ -327,12 +327,22 @@ def set_config(bot, trigger):
         bot.say(exc.args[1])
         return
 
+    # Get a descriptor class for the option if it's a static section
+    descriptor = getattr(section.__class__, option) if static_sec else None
+
     # Display current value if no value is given
     if not value:
-        if option.endswith("password") or option.endswith("pass"):
+        value = getattr(section, option)
+
+        if descriptor is not None:
+            if getattr(descriptor, 'is_secret', False):
+                # Keep secret option as secret
+                value = "(password censored)"
```

Comment:
```suggestion
                value = "(secret value censored)"
```

It's not necessarily a password, unlike the "dumb" name-heuristic fallback.